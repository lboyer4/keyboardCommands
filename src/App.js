import React, { Component } from 'react';
import './SCSS/App.scss';

class App extends Component {
  constructor() {
    super()
    this.state = {
        allCards=[],
        level: '',
        practiceCards=[],
        wrongCards=[],
        correctCards=[]
    }

  }
  render() {
    return (
      <div>

      </div>
      )
  }
}

export default App;
