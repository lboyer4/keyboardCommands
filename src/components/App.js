import React, { Component } from 'react';
import '../SCSS/_App.scss';
import StartHolder from './StartHolder.js';
import Cloud from '../images/Cloud-One.png';
import keyCommands from '../key-command-data.js';
import CardHolder from './CardHolder.js';
import Directory from './Directory.js';

class App extends Component {
  constructor() {
    super()
    this.state = {
        allCards: keyCommands || [],
        level: '',
        currentCards: [],
        currentCard: {},
        incorrectCards: [],
        score: 0
    }
  }
  
  setLevel = (selectedLevel) => {
      this.setState({level: selectedLevel}) 
      this.setCurrentCards(this.state.allCards, selectedLevel); 
      this.setCurrentCard() 
  }

  setCurrentCards = (allCards, selectedLevel) => {
    let all = this.state.allCards
    let current = this.state.currentCards
    return all.filter((card) => {
      if(card['level'] === selectedLevel) {
        this.state.currentCards.push(card)
      }
    })
  }

  setWrongCards = (incorrectCards) => {
    console.log(this.state)
    JSON.parse(localStorage.getItem('storedState'))
    console.log('storedState')
    this.setState({
      currentCard: this.state.incorrectCards.shift()
    })
      if (this.state.incorrectCards.length) {
        this.setState({level: 'Practice'})
      }
  }

  setCurrentCard = () => {
    if (this.state.level === 'Practice' && this.state.incorrectCards.length) {
      this.setWrongCards();
    } else {
    this.setState({currentCard: this.state.currentCards.shift()}) 
    if (!this.state.currentCards.length) {
      this.state.level = ''
    } 
  }
    this.setLocalStorage()   
  }

  setLocalStorage = () => {
    localStorage.setItem('storedState', JSON.stringify(this.state))
  }

  // getLocalStorage = () => {
  //   localStorage.getItem('storedState', JSON.stringify(this.state))
  // }

  setIncorrectStack = () => {
    let wrongAnswers = this.state.incorrectCards
    wrongAnswers.push(this.state.currentCard)
  }

  updateScore = () => {
    this.setState({score: this.state.score += 1})
  }

  clearAll = () => {
    this.setState({ 
        allCards: keyCommands || [],
        level: '',
        currentCards: [],
        currentCard: {},
        incorrectCards: [],
        score: 0
    })
     this.setLocalStorage()
  }

  returnStartPage = () => {
    this.setState({
      level: ''
    })
  }


  render() {
    // console.log(this.state.currentCards)
    // console.log(this.state.incorrectCards)
    let startGame = <StartHolder 
      setLevel={this.setLevel}
      setWrongCards={this.setWrongCards}
      incorrectCards={this.state.incorrectCards}
       />

    let gameBegin = <CardHolder 
      updateScore={this.updateScore}
      setCurrentCard={this.setCurrentCard}
      setIncorrectStack={this.setIncorrectStack}
      currentCard={this.state.currentCard}
      level={this.state.level}
      />

    let scoreBoard = <Directory 
      score={this.state.score} 
      clearAll={this.clearAll} 
      returnStartPage={this.returnStartPage} 
      />

    let display = (this.state.level) ? gameBegin : startGame

    let header = this.state.score && scoreBoard
    return ( 
      
      <main className="App">
        <header className="header">
          {header}
        </header>
        <img className="cloud-one" src={Cloud} width="200px" alt="beautiful magic cloud" />
        
        {display}
        
      </main>
    )
  }
}

export default App;
