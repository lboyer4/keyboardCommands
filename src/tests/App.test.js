import React from 'react';
import ReactDOM from 'react-dom';
import App from '../components/App';
import { shallow } from 'enzyme'; 


describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <App />
    );
  });

  it("should match the snapshot with all data passed in", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("update score method", () => {
		it("should update the score by one", () => {
			expect(wrapper.state().score).toEqual(0)
			wrapper.instance().updateScore();
			expect(wrapper.state().score).toEqual(1)
		});
  });

  describe("clear all method", () => {
  	it("should return state to original values", () => {
  		wrapper.setState({
  			allCards: ['one', 'two'],
        level: 'foobar',
        currentCards: ['3'],
        currentCard: {test: 'cardddds'},
        incorrectCards: ['four', 'five'],
        score: 5
    	});
    	wrapper.instance().clearAll();
    	expect(wrapper.state()).toEqual({        
    		allCards: [],
        level: '',
        currentCards: [],
        currentCard: {},
        incorrectCards: [],
        score: 0
    	});
  	});
  });

  describe("return Start Page Method", () => {
  	it("should return to the start page when level is empty", () => {
  		wrapper.setState({level: 'Beginner'});
  		expect(wrapper.state().level).toEqual('Beginner');
  		wrapper.instance().returnStartPage();
  		expect(wrapper.state().level).toEqual('');
  	});
  });
  	
  describe("Set incorrect Stack Method", () => {
  	it("should set the incorrect cards state", () => {
  		wrapper.setState({currentCard: {test: 'one'}})
  		expect(wrapper.state().currentCard).toEqual({test: 'one'});
  		expect(wrapper.state().incorrectCards).toEqual([]);
  		wrapper.instance().setIncorrectStack();
  		expect(wrapper.state().incorrectCards).toEqual([{test: 'one'}]);
  	})
  })
});





































// const mockData =  [{
// 		"task": "Copy",
// 		"command": "command C",
// 		"level": "Beginner"
// 	},

// 	{
// 		"task": "Paste",
// 		"command": "command P",
// 		"level": "Beginner"
// 	},

// 	{
// 		"task": "Cut",
// 		"command": "command X",
// 		"level": "Beginner"
// 	},

// 	{
// 		"task": "Save",
// 		"command": "command S",
// 		"level": "Beginner"
// 	}];

// 	const mockIncorrectCards = [

// 		{
// 		"task": "Cut",
// 		"command": "command X",
// 		"level": "Beginner"
// 	},
// 	{
// 		"task": "Save",
// 		"command": "command S",
// 		"level": "Beginner"
// 	}
// 	];



// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });

// describe("App", () => {
// 	let wrapper;

// 	beforeEach(() => {
// 		wrapper = shallow(
// 			<App />
// 		);
// 	});

// 	it("should match snapshot", () => {
// 		expect(wrapper).toMatchSnapshot();
// 	});

//  	it("should have a proper default state", () => {
//   	expect(wrapper.state()).toEqual({
// 	   	allCards: [],
// 	    level: '',
// 	    currentCards: [],
// 	    currentCard: {},
// 	    incorrectCards: [],
// 	    score: 0    
//   	});
//  	});
  
// 	describe("Set Level", () => {
// 		it("should set the state of level", () => {
// 			expect(wrapper.state().level).toEqual('');
// 			wrapper.instance().setLevel(
// 				"Advanced");
// 			expect(wrapper.setState({level:'Advanced'}))
// 		});

// 	// it("should call setCurrentCard", () => {

// 	// 	expect(mockSetCurrentCards).toBeCalled()
// 	// 	})
// 	// });
// 	});


// 	describe("Should set current cards", () => {

// 		it("should set the current cards", () => {
// 			expect(wrapper.state().currentCards).toEqual([])
// 			wrapper.instance().setCurrentCards(mockData)
// 			expect(wrapper.state().currentCards).toEqual(mockData)
// 		});
// 	});

// 	describe("it should set wrong cards", () => {

// 		it('should set level to Practice', () => {
// 			expect(wrapper.state().level).toEqual("");
// 			wrapper.instance().setWrongCards(mockIncorrectCards)
// 			expect(wrapper.state().level).toEqual('Practice');
// 		});
// 	});

// });

