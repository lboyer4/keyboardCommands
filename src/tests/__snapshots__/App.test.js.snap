// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in 1`] = `
<main
  className="App"
>
  <header
    className="header"
  >
    <Directory
      clearAll={[Function]}
      incorrectCards={Array []}
      returnStartPage={[Function]}
      score={0}
    />
  </header>
  <img
    alt="beautiful magic cloud"
    className="cloud-one"
    src="Cloud-One.png"
    width="200px"
  />
  <StartHolder
    incorrectCards={Array []}
    setLevel={[Function]}
    setWrongCards={[Function]}
  />
</main>
`;
